pkg/kubevirt/core/core.go

```diff
@@ -248,10 +234,12 @@ func (p PluginSPIImpl) CreateMachine(ctx context.Context, machineName string, pr
 		},
 	}
 
-	if providerSpec.MemoryFeatures != nil && providerSpec.MemoryFeatures.Hugepages != nil {
-		virtualMachine.Spec.Template.Spec.Domain.Memory = &kubevirtv1.Memory{
-			Hugepages: providerSpec.MemoryFeatures.Hugepages,
-		}
+	if providerSpec.CPU != nil {
```

Comment:
Why do you need this check, couldn't you just set it in the struct literal above?